pokemongo_bot/cell_workers/recycle_items.py

```diff
@@ -1,85 +1,185 @@
+        }
+      }
+    }
+    """
+
     def initialize(self):
+        self.items_filter = self.config.get('item_filter', {})
         self.min_empty_space = self.config.get('min_empty_space', None)
-        self.item_filter = self.config.get('item_filter', {})
         self._validate_item_filter()
 
     def _validate_item_filter(self):
+        """
+        Validate user's item filter config
+        :return: Nothing.
+        :rtype: None
+        :raise: ConfigException: When an item doesn't exist in ../../data/items.json
+        """
         item_list = json.load(open(os.path.join(_base_dir, 'data', 'items.json')))
-        for config_item_name, bag_count in self.item_filter.iteritems():
+        for config_item_name, bag_count in self.items_filter.iteritems():
             if config_item_name not in item_list.viewvalues():
                 if config_item_name not in item_list:
-                    raise ConfigException(
-                        "item {} does not exist, spelling mistake? (check for valid item names in data/items.json)".format(
-                            config_item_name))
+                    raise ConfigException("item {} does not exist, spelling mistake? (check for valid item names in data/items.json)".format(config_item_name))
+
+    def should_run(self):
+        """
+        Returns a value indicating whether the recycling process should be run.
+        :return: True if the recycling process should be run; otherwise, False.
+        :rtype: bool
+        """
+        if inventory.items().get_space_left() >= (DEFAULT_MIN_EMPTY_SPACE if self.min_empty_space is None else self.min_empty_space):
+            return False
+        return True
 
     def work(self):
-        items_in_bag = self.bot.get_inventory_count('item')
-        total_bag_space = self.bot.player_data['max_item_storage']
-        free_bag_space = total_bag_space - items_in_bag
-
-        if self.min_empty_space is not None:
-            if free_bag_space >= self.min_empty_space:
-                    self.emit_event(
-                        'item_discard_skipped',
-                        formatted="Skipping Recycling of Items. {space} space left in bag.",
-                        data={
-                            'space': free_bag_space
-                        }
-                    )
-                    return
-
-        self.bot.latest_inventory = None
-        item_count_dict = self.bot.item_inventory_count('all')
-
-        for item_id, bag_count in item_count_dict.iteritems():
-            item_name = self.bot.item_list[str(item_id)]
-            id_filter = self.item_filter.get(item_name, 0)
-            if id_filter is not 0:
-                id_filter_keep = id_filter.get('keep', 20)
+        """
+        Discard items if necessary.
+        :return: Always returns WorkerResult.SUCCESS.
+        :rtype: WorkerResult
+        """
+        # Updating inventory
+        inventory.init_inventory(self.bot)
```

Comment:
Inventory is already initialized when the API is set up, you don't need this here. This will ready all JSON cache files again and can slow down the tick a lot.